### ISSUE TITLE
Support change tracking and lazy-loading proxies for complex types

### ISSUE THREAD

**Issue by ajcvickers** [dotnet/efcore#31634]

**Comment by cpa-level-it** [comment]
Is it correct that without this it'll not be possible to use complex types with `UseLazyLoadingProxies` ? 

Using both right now I have the following issue when I do a `Find` on a DbSet.

The foreach throws a `System.NotImplementedException: 'The method or operation is not implemented.'
` exception in `EntityMaterializerSource.cs`

```C#
private InstantiationBinding ModifyBindings(ITypeBase structuralType, InstantiationBinding binding)
{
	InstantiationBindingInterceptionData interceptionData = new InstantiationBindingInterceptionData(structuralType);
	foreach (IInstantiationBindingInterceptor bindingInterceptor in _bindingInterceptors)
	{
		binding = bindingInterceptor.ModifyBinding(interceptionData, binding);
	}
	return binding;
}
```
Without the call to `UseLazyLoadingProxies` the `Find` correctly returns the value.

Will this be addressed before the release of the 8.0 version ?

Thanks

**Comment by oleg-varlamov** [comment]
We had the same problem - it is impossible to use complex properties if `UseLazyLoadingProxies `is used in the project. Unfortunately, without fixing this problem, we cannot move to using complex properties.
